[Blog] Feature: About the Author

Lezette opened the pull request:
**Description**

This PR fixes #2780

**Notes for Reviewers**


**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

Lezette commented:
@leecalcote  @Nikhil-Ladha I am trying to add the "share post" feature, But I don't know if there's a way to do it currently without installing a package. Do you have any suggestions on how to do this?

l5io commented:
🚀 Preview for commit f338c9471a826305025c36729194f7f9d26d23c6 at: https://62a0f62a6f3f4e233023259c--layer5.netlify.app

Nikhil-Ladha commented:
> @leecalcote @Nikhil-Ladha I am trying to add the "share post" feature, But I don't know if there's a way to do it currently without installing a package. Do you have any suggestions on how to do this?

Considering we actually want a direct share feature, using a package is the only way ahead.

Also, please follow the contributing guidelines and give a meaningful name to your PR and a description. 🙏🏼

leecalcote commented:
"share post" meaning "about the author", right?

leecalcote commented:
> "share post" meaning "about the author", right?

Oh! Now, I understand. Yes, you're looking for a solution to make the three social share buttons hyperlinks that will share a new post on the relevant social network. This could be a custom construction for each social network, like this for Twitter:

https://twitter.com/intent/tweet?url=Evolution%20of%20the%20Meshery%20CLI%20Command%20Reference%20https%3A%2F%2Flayer5.io%2Fblog%2Fmeshery%2Fevolution-of-the-meshery-cli-command-reference

leecalcote reviewed (changes requested):
@Lezette, whoa! This is looking pretty good. 

Open issues:
1. Dynamic construction of social shares (or use a package)
1. Mobile view should display this About the Author, but it currently goes missing (see the mockup for responsive design).
1. Please note the CSS issue of profile photos not being clipped (photos overlapping the circular outline). Example: 
<img width="271" alt="Screen Shot 2022-06-08 at 6 49 48 PM" src="https://user-images.githubusercontent.com/7570704/172736807-091c22e4-a7bf-441c-9311-dc33a6e240d8.png">

Lezette commented:
@leecalcote Noted, thanks

leecalcote commented:
> @leecalcote Noted, thanks

@Lezette what approach to the share buttons are you exploring?

Lezette commented:
> > @leecalcote Noted, thanks
> 
> @Lezette what approach to the share buttons are you exploring?

@leecalcote I'll be using a package, I'm checking out [React share](https://github.com/nygardk/react-share)

Lezette commented:
@leecalcote

- I've added the share post functionality
- I couldn't replicate the  CSS issue of profile photos not being clipped but I added a fix for that as well, can you please confirm if it's still the same
- I couldn't find the mockup for mobile in the Figma file but I'm showing it on mobile now (see image below)

<img width="340" alt="image" src="https://user-images.githubusercontent.com/42512400/173083844-27c74004-b747-47cf-ad90-5e0bfb9de837.png">

l5io commented:
🚀 Preview for commit db4c3e77d042ca2088145701020a73d6976a2626 at: https://62a351b2ad366245054f5168--layer5.netlify.app

leecalcote commented:
Example of the overlap issue from the latest build.
<img width="1388" alt="Screen Shot 2022-06-10 at 9 20 33 AM" src="https://user-images.githubusercontent.com/7570704/173086124-101a4e78-b9d6-49be-beb8-486cd235d430.png">

Nikhil-Ladha commented:
Also, regarding the positioning of the card I wonder if you could make it position as in the mockup shown at #2848 . That would avoid an extra column on the right of the page.

Nikhil-Ladha commented:
The Twitter link opens up in the app (mobile), but the same doesn't happen for LinkedIn, can you also check that once?

Lezette commented:
@leecalcote I can see it now, it happens on safari, I'll fix that now

Lezette commented:
@Nikhil-Ladha I was trying to work on that but based on my knowledge it would require a column layout and I was having issues with responsiveness.  Any suggestions on how to achieve that? Also for the LinkedIn do you mean it doesn't the link?

leecalcote commented:
> 

Shoot. I see now that when designing, I gave up on the notion of a responsive About the Authors section. Ok, maybe something to consider in the future, but for now, to @Nikhil-Ladha's point, Acceptance Criteria #1 stipulates that the About the Authors sections disappears.

Lezette commented:
@leecalcote I'm sorry I'm a bit confused, do you mean I should hide About the Authors section on mobile now?

leecalcote commented:
@Lezette, the confusion is my fault. Let me remove myself from this discussion and allow the person who is providing clarity, @Nikhil-Ladha, to carry this forward with you. I'll let @Nikhil-Ladha make this judgement call (and any other on this PR).

Nikhil-Ladha commented:
> @Nikhil-Ladha I was trying to work on that but based on my knowledge it would require a column layout and I was having issues with responsiveness. Any suggestions on how to achieve that?

Not necessarily, but rather a `float: right` (or maybe flexbox?) on the author card would do.

> Also for the LinkedIn do you mean it doesn't the link?

Try opening the above preview on your mobile, and clicking the LinkedIn share button on the blog, you will notice that it opens up the post on a browser and not on the LinkedIn app (if you have it installed), but for Twitter, it opens up on the Twitter app directly. I don't have Facebook, so I can't say about that, but if you have, maybe you can also test that.

Nikhil-Ladha commented:
> @leecalcote I'm sorry I'm a bit confused, do you mean I should hide About the Authors section on mobile now?

Yes, @Lezette for now. Let's make the author card hidden in mobile view.

But for share, we can use the classic single share button like 
![image](https://user-images.githubusercontent.com/29438875/173103318-dd07b760-4c2e-44a2-9818-d8b08450254e.png)
that would allow the sharing on mobile devices. Now, I am not sure how flexible that is, you can give it a try and if it is a lot you can ignore it for now and take up in a later PR.

Lezette commented:
> > @Nikhil-Ladha I was trying to work on that but based on my knowledge it would require a column layout and I was having issues with responsiveness. Any suggestions on how to achieve that?
> 
> Not necessarily, but rather a `float: right` (or maybe flexbox?) on the author card would do.
> 
> > Also for the LinkedIn do you mean it doesn't the link?
> 
> Try opening the above preview on your mobile, and clicking the LinkedIn share button on the blog, you will notice that it opens up the post on a browser and not on the LinkedIn app (if you have it installed), but for Twitter, it opens up on the Twitter app directly. I don't have Facebook, so I can't say about that, but if you have, maybe you can also test that.

@Nikhil-Ladha thanks, I used float but it was not giving the desired result and it wasn't making it responsive. I am currently using flexbox.

For the share button on mobile, I'm trying to checkout a way to do that, Also the issue with the LinkedIn not opening the app is from the package I am currently using, I'll checkout another package that can fix that.

l5io commented:
🚀 Preview for commit f227e49136e2b7b79b45b116cae6cd8e9ae339e1 at: https://62a4bfdeaf906f39a7619d29--layer5.netlify.app

debo19 commented:
@Lezette lets discuss your updates on the websites call tomorrow 🚀

leecalcote commented:
@Lezette heads-up: merge conflict to resolve.

Nikhil-Ladha commented:
@Lezette any updates on this?

l5io commented:
🚀 Preview for commit 6aae0c918edbdd57bf909322878f5acb86fade86 at: https://62baeb92eee5e11a18b630a2--layer5.netlify.app

leecalcote commented:
Something like this is another route: 

`clip-path: circle(60px at center);`

Lezette commented:
> clip-path: circle(60px at center)

Thank you I'll do that now

debo19 commented:
@Lezette were you successful here? Merge conflicts to address too.

leecalcote commented:
@Lezette merge conflict...

debo19 commented:
@Lezette I've tried to help a bit here. I think this fixes the problem in safari. FYI, I've created a separate file for this component. There might be some responsiveness issues remaining. I hope you can take this forward from here. Since this feature links to MeshMap, we need this published as soon as possible.

l5io commented:
🚀 Preview for commit a2bd74f6b857aec6615f91bedcb8cdd35c1cb185 at: https://62c4b8a57f26e53096db2c1d--layer5.netlify.app

Lezette commented:
Thanks I’ll take it up from there

> On 5 Jul 2022, at 23:20, l5io ***@***.***> wrote:
> 
> ﻿
> 🚀 Preview for commit a2bd74f at: https://62c4b8a57f26e53096db2c1d--layer5.netlify.app
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.

Lezette commented:
@debo19 I have made the fixes to the responsiveness issue. Thank you so much for your help.

@Nikhil-Ladha I have added the share icon on mobile, I didn't know exactly where to place the icon so I placed it beside the tags, Once the user clicks on the share icon it shows a popup for the share buttons. Please let me know what you think.

https://user-images.githubusercontent.com/42512400/177588831-b25a5cea-4536-4485-abb9-1c5fb1d09158.mov

l5io commented:
🚀 Preview for commit 9b6a86384d3550c6e8b2c811fc38c6c5078d995e at: https://62c5af81cf30bc1b6e83996d--layer5.netlify.app

Nikhil-Ladha commented:
Ok, it seems like this doesn't suit our needs well.
Let's update it to have a text like `Share on Fb, li, twt, link`, and place it just above the `Tags` section.
All 3 social media icons and the last one would be copy to clipboard icon which will copy the link to the page.

debo19 commented:
@Lezette merge conflicts to address...

l5io commented:
🚀 Preview for commit 85768b72ef4bc62a9554898bde24c586116136d6 at: https://62d0a0f83c4f1015c6507a21--layer5.netlify.app

leecalcote reviewed (approved):
I don't know if it is because of the slowness of Netlify, but the Related Blogs section isn't updating responsively unless I refresh my browser. I assume this isn't an issue cause by this PR, but is a Netlify-slowness issue.

Otherwise, the About the Authors looks great in Safari and the social share links each work well. 👏 

No About the Authors shown when there isn't a community member profile or when there is more than one author or the author is "Layer5 Team" - that makes sense.

About the Authors is shown whether the community member is active or inactive. 👍

leecalcote commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+            overflow: hidden;
+            a{
+                img{
+                    clip-path: circle(50% at 50% 50%);
```
Cool. 🙂

debo19 reviewed (changes requested):
<img width="468" alt="Screenshot 2022-07-15 at 2 38 48 AM" src="https://user-images.githubusercontent.com/85789734/179186102-7fb62908-3d95-469d-adaa-b2425e830221.png">
This shouldn't be visible on desktop view.

Lezette commented:
> <img alt="Screenshot 2022-07-15 at 2 38 48 AM" width="468" src="https://user-images.githubusercontent.com/85789734/179186102-7fb62908-3d95-469d-adaa-b2425e830221.png">
> 
> This shouldn't be visible on desktop view.

Done @Nikhil-Ladha

l5io commented:
🚀 Preview for commit 927fa3ebefc850b1472f103bf1c300b952619f95 at: https://62d15a3a8083e10afb2eaa85--layer5.netlify.app

leecalcote commented:
Good eye, @debo19 😄

Nikhil-Ladha commented:
Make `Share Post` `h4`, and the div width for social media icons can be reduced to somewhat around ~40%, which makes them closer.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+                font-weight: 500;
+                font-size: 24px;
+                line-height: 32px;
+                color: #1E2117;
+            }
+            .share-icons-container{
+                width: 60%;
+                margin: 0 auto;
+                display: flex;
+                justify-content: space-between;
+            }
+        }
+    }
+}
+
+
+ @media (max-width: 975px) { 
+     .author-info-section{
+       width: 30%;
+     }
+ }
+ @media (max-width: 770px) { 
+     .author-info-section{
+       display: none;
+     }
+ }
+
+ @media screen and (max-width: 360px){
+    .author-info-section{
+        width: 90%;
+        margin: 0 auto;
+         .authors-info-container{
+            .authors-head-shot{
+                width: 40%;
+                height: 40%;
+            }
+         }
+     }
+}
+`;
```
Move this to `author.style.js` file (create a new one)

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+                <Image {...authorInformation?.frontmatter?.image_path} imgStyle={{ objectFit: "cover" }} alt={authorInformation.frontmatter?.name} className="authors-image" />
+              </Link>
+            </div>
+            <h5>{authorInformation.frontmatter?.name}</h5>
```
Make it `h4`

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+              {authorInformation.frontmatter?.bio}
+            </p>
+            <div className="authors-info-meshery">
+              <h6>MeshMap</h6>
```
`h5`

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+        text-align: center;
+        margin-bottom: 3rem;
+        h3{
+            font-size: 20px;
```
Prefer `rem`, i.e, this should be `1.5rem`

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+        margin-bottom: 3rem;
+        h3{
+            font-size: 20px;
+            line-height: 32px;
+            font-weight: 500;
+            color: #1E2117
```
These styles are already defined globally, no need to redefine here

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.js:
```diff
@@ -0,0 +1,188 @@
+        }
+        .share-section{
+            p{
+                font-family: 'Qanelas Soft';
```
`p` tags use `Open Sans`, and it is defined globally. Not needed here.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/blogSingle.style.js:
```diff
@@ -36,6 +48,33 @@ const BlogPageWrapper = styled.div`
+                justify-content: space-between;
+                align-items: center;
+                padding: 0 .5rem;
+                border-radius: 10px;
+                font-size: 30px;
```
Use `rem`

Nikhil-Ladha commented on src/sections/Blog/Blog-single/index.js:
```diff
@@ -9,90 +9,146 @@ import RelatedPosts from "../../../components/Related-Posts";
+            />
+            <div className="post-tag-container">
+              <div className="post-share-mobile">
+                <p>Share Post:</p>
```
Make it `h4`

Nikhil-Ladha reviewed (changes requested):
No need to redefine the `font-family` for anything, it is defined globally and shouldn't be changed.
Also, for `p` and `h*` tags only update `font-size`, and `font-weight` (that too only when necessary) rest of the styles are defined globally.

Nikhil-Ladha commented:
Use this icon for the copy to clipboard `IoIosCopy`.
Also, add a tooltip for `Copied` when the user clicks on the icon. I believe the npm package should provide that feature.

leecalcote commented:
@Lezette be aware of the merge conflict that has popped up.

Lezette commented:
@Nikhil-Ladha I have made the updates, For the tooltip, Please check out the video demo below, Is the style fine? Also, I made the tooltip disappear after 3 seconds. Please let me know what you think?


https://user-images.githubusercontent.com/42512400/179517747-a262ebbb-6487-4b47-8d73-8d74123689bc.mov

l5io commented:
🚀 Preview for commit fb560d0ff81d4125659b437bb7cf5ea02b5066aa at: https://62d55da981fa5c4977b8a748--layer5.netlify.app

Nikhil-Ladha commented:
Position the tooltip to the right/bottom of the icon.

Nikhil-Ladha reviewed (commented):
You can use `h4` for both author name and `MeshMap` heading and set the fon-size to `1.25rem` for both.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.style.js:
```diff
@@ -0,0 +1,120 @@
+        }
+        p{
+            font-weight: 500;
+            font-size: 15px;
```
Make it `1rem`.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.style.js:
```diff
@@ -0,0 +1,120 @@
+                color: #1E2117;
+            }
+            p{
+                font-weight: 500;
```
Same as above

Nikhil-Ladha commented on src/sections/Blog/Blog-single/author.style.js:
```diff
@@ -0,0 +1,120 @@
+            p{
+                font-weight: 500;
+                font-size: 15px;
+                line-height: 22px;
```
Please use `rem`
Make it `1.25rem`
Update in other places too.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/index.js:
```diff
@@ -9,90 +9,159 @@ import RelatedPosts from "../../../components/Related-Posts";
+
+  const authorInformation = authors.nodes.filter((author) => author.frontmatter.name === frontmatter.author)[0];
+
+  const shareQuote = `Check out this post from layer5 ${frontmatter.title}`;
```
Add double quotes around `${frontmatter.title}`

l5io commented:
🚀 Preview for commit 7392d913d2b7d840ddef76fa05ad79e072ae179e at: https://62d56dae59ccde0069a81a7d--layer5.netlify.app

Lezette commented:
I've fixed the updates @Nikhil-Ladha

Nikhil-Ladha commented:
The icons in `Share Post` are not vertically center-aligned in mobile view.
Also, make the `font-size` of `p` `1rem` as suggested above. The default is `1.125rem` and give a `line-height` of `1.5rem`.

Nikhil-Ladha commented on src/sections/Blog/Blog-single/blogSingle.style.js:
```diff
@@ -36,6 +48,43 @@ const BlogPageWrapper = styled.div`
+            .share-icons-container {
+                display: flex;
+                margin: auto 0;
+                justify-content: space-between;
```
Both under "MeshMap" and "About the Author"

Nikhil-Ladha reviewed (approved):
Looking real good now @Lezette .
Great work 🥇

l5io commented:
🚀 Preview for commit 0436c78de3883b49195ffe6fcb99258a10738cbb at: https://62d64307f767f545dc5fc8e3--layer5.netlify.app

debo19 commented:
Yay finally 🎉 Great work @Lezette
